src/firewall/firewalld.rs

```diff
@@ -1,6 +1,6 @@
 use crate::firewall;
-use crate::network::types;
-use crate::network::types::{Network, PerNetworkOptions, TeardownPortForward};
+use crate::network::internal_types::{SetupPortForward, TearDownNetwork, TeardownPortForward};
+use crate::network::{internal_types, types};
```

Comment:
This seems redundant, you're importing everything in internal_types - and then some specific bits of it as well, above?